### PR TITLE
cnet: fix some init issues and cleanup IPv6 ifdefs

### PR DESCRIPTION
**Pull request by KeithWiles**
- Add test case to check for ipv6 in graph patterns
- Remove some CNET_ENABLE_IP6 ifdefs and add the test at runtime, the compile will remove the code based on CNET_ENABLE_IP6 value.
- Missing initialization for route6 and nd6 in cnet_config_create()
- Remove some dead code in cnet_protosw.c
- Change ip6_get_flowlable, ip6_get_tclass and ip6_get_version routines to static inline, because when building use debug mode the compiler would flag this as an error.
- fixed a couple comment typos
- fix up pkt_ctrl.c code to link IPv6 routines to the Rx/Tx output routines. Verified by running CNET cli command 'gdot' and then use xdot program to display the cnet_2.dot file in top level directory.
- Make sure protosw contains the UDP and TCP entries for IPv6. Plus a few bugs were found because of IPv6 was added.

### CODE REVIEW

File: lib/cnet/cnet/cnet.c [hunk 1]
```diff
@@ -77,8 +78,14 @@ cnet_config_create(uint32_t num_chnls, uint32_t num_routes)
         if (cnet_route4_create(cnet, num_routes, 0) < 0)
             CNE_ERR_GOTO(leave, "Failed to create route\n");
 
+        if (cnet_route6_create(cnet, num_routes, 0) < 0)
```
Comment (maryamtahhan) [inline lib/cnet/cnet/cnet.c:81]: do we need to check if (CNET_ENABLE_IP6) before we call `cnet_route6_create()`

**Comment by maryamtahhan** [comment]
Will try to review this tomorrow.

File: lib/cnet/icmp6/cnet_icmp6.c [hunk 2]
```diff
@@ -30,8 +30,6 @@ icmp6_create(void *_stk)
 
     CNE_ASSERT(psw != NULL);
 
-    cnet_ipproto_set(IPPROTO_ICMPV6, psw);
```
Comment (mnjdhl) [inline lib/cnet/icmp6/cnet_icmp6.c:33]: Any idea why this is not needed for ICMPV6 ?

File: lib/cnet/udp/udp_output.c [hunk 3]
```diff
@@ -55,10 +55,7 @@ udp_output_header(pktmbuf_t *m, uint16_t nxt)
     udp->dgram_cksum = 0;
 
 #if CNET_ENABLE_IP6
```
Comment (mnjdhl) [inline lib/cnet/udp/udp_output.c:57]: This ifdef can be removed as same condition has been added in is_pcb_dom_inet6().

**Review by mnjdhl — CHANGES_REQUESTED** [review]
Please look at ifdef at line#57 in lib/cnet/udp/udp_output.c.

Looks like it is not needed.

File: lib/cnet/udp/udp_output.c (hunk 3 above)
Comment (KeithWiles) [reply lib/cnet/udp/udp_output.c:57]: Removed the ifdef

**Comment by KeithWiles** [comment]
The ifdef in cnet_output.c and cnet_tcp_output.c are required, at least I could not see an easy way remove them and have defines available for the nxt variable. :-(

File: lib/cnet/cnet/cnet.c [hunk 4]
```diff
@@ -120,6 +120,7 @@ cnet_stop(void)
         cnet_drv_destroy(cnet);
         cnet_route4_destroy(cnet);
```
[the remark this answers is not in the document]
Comment (mnjdhl) [reply lib/cnet/cnet/cnet.c:121]: Similalry icmp6_destroy(), ipv4_destroy and ipv6_destroy are also missing.

**Review by mnjdhl — CHANGES_REQUESTED** [review]
Pl look at my comment on changes in cnet.c

File: lib/cnet/protosw/cnet_protosw.c [hunk 5]
```diff
@@ -14,12 +14,6 @@
 #include "cnet_protosw.h"
 #include "cne_log.h"        // for CNE_LOG, CNE_LOG_DEBUG, CNE_LOG_WARNING
 
-/**
```
Comment (maryamtahhan) [inline lib/cnet/protosw/cnet_protosw.c:17]: I think this is the file description? maybe we didn't mean to delete it?

Comment (maryamtahhan) [reply lib/cnet/protosw/cnet_protosw.c:17]: or maybe it is just for the array?

**Review by maryamtahhan — APPROVED** [review]
Looks great overall. just a couple of minor nits. 

Thanks for doing this @KeithWiles

File: lib/cnet/cnet/cnet.c (hunk 1 above)
Comment (KeithWiles) [reply lib/cnet/cnet/cnet.c:81]: Moved the check into the cnet_route6_create() function.

File: lib/cnet/protosw/cnet_protosw.c (hunk 5 above)
Comment (KeithWiles) [reply lib/cnet/protosw/cnet_protosw.c:17]: Yes you are correct the comment is for the file.

**Comment by KeithWiles** [comment]
@mnjdhl please approve the PR if you want(I believe you can approve or just a message you are OK with the code) and when you have completed your review. Thanks

File: lib/cnet/cnet/cnet.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (mnjdhl) [reply lib/cnet/cnet/cnet.c:121]: Looks good

**Review by mnjdhl — APPROVED** [review]
Thanks a lot Keith for the wonderful contribution.

**Pull request merged**